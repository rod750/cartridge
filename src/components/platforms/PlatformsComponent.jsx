import React from 'react';
import { Query } from 'react-apollo';
import { GET_PLATFORMS_QUERY } from '../../queries/platforms/getPlatforms';
import { PlatformComponent } from './PlatformComponent';
import { Notification, Icon } from 'bloomer';

const PlatformItem = (platform) => {
    return <PlatformComponent platform={platform} key={platform.id} />;
}

export class PlatformsComponent extends React.Component {
    render() {
        return (
            <Query query={GET_PLATFORMS_QUERY}>
                {({data, loading}) => {
                    if(loading) {
                        return <Notification><Icon className="mdi mdi-loading spinner" /> Loading</Notification>;
                    }
                    if(!data || !data.platforms) {
                        return <Notification isColor="dark"><Icon className="mdi mdi-alert" />Uh oh. Something is not working.</Notification>;
                    }
                    if(data.platforms.length === 0){
                        return <Notification>There are no platforms registered.</Notification>;
                    }

                    return data.platforms.map(PlatformItem);
                }}
            </Query>
        );
    }
}
