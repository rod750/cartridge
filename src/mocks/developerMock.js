export const developerMock = {
    id: 1,
    name: "Developer test"
};
