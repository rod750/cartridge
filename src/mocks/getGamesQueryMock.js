import { GET_GAMES_QUERY } from '../queries/games/getGames';

export const getGamesQueryMock = {
    request: {
        query: GET_GAMES_QUERY
    },
    results: {
        data: {
            games: [
                {
                    id: 1,
                    name: "Testing",
                    tags: [
                        { id: 1, name: "Test tag" }
                    ],
                    platform: {
                        id: 1,
                        name: "Test platform"
                    },
                    developers: [
                        { id: 1, name: "Test developer" }
                    ]
                }
            ]
        }
    }
};
