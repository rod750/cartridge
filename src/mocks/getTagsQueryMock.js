import { tagMock } from "./tagMock";
import { GET_TAGS_QUERY } from "../queries/tags/getTags";

export const getTagsQueryMock = {
    request: {
        query: GET_TAGS_QUERY
    },
    results: {
        data: {
            tags: [
                tagMock
            ]
        }
    }
};
