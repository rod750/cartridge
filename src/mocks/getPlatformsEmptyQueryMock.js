import { GET_PLATFORMS_QUERY } from '../queries/platforms/getPlatforms';

export const getPlatformsEmptyQueryMock = {
    request: {
        query: GET_PLATFORMS_QUERY
    },
    results: {
        data: {
            platforms: []
        }
    }
};
