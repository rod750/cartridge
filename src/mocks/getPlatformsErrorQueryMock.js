import { GET_PLATFORMS_QUERY } from '../queries/platforms/getPlatforms';

export const getPlatformsErrorQueryMock = {
    request: {
        query: GET_PLATFORMS_QUERY
    },
    results: {
        data: undefined
    }
};
