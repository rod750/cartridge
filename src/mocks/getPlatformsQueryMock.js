import { GET_PLATFORMS_QUERY } from '../queries/platforms/getPlatforms';
import platformMock from './platformMock';

export const getPlatformsQueryMock = {
    request: {
        query: GET_PLATFORMS_QUERY
    },
    results: {
        data: {
            platforms: [
                platformMock
            ]
        }
    }
};
