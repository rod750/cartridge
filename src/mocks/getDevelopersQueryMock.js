import { GET_DEVELOPERS_QUERY } from "../queries/developers/getDevelopers";
import { developerMock } from "./developerMock";

export const getDevelopersQueryMock = {
    request: {
        query: GET_DEVELOPERS_QUERY
    },
    results: {
        data: {
            developers: [
                developerMock
            ]
        }
    }
};
