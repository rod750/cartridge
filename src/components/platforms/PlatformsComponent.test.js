import React from 'react';
import ReactDOM from 'react-dom';
import { PlatformsComponent } from './PlatformsComponent';
import { getPlatformsQueryMock } from '../../mocks/getPlatformsQueryMock';
import { MockedProvider } from 'react-apollo/test-utils';
import renderer from 'react-test-renderer'
import { getPlatformsEmptyQueryMock } from '../../mocks/getPlatformsEmptyQueryMock';
import { getPlatformsErrorQueryMock } from '../../mocks/getPlatformsErrorQueryMock';
const wait = require('waait');

const mocks = [
    getPlatformsQueryMock
];

const emptyMocks = [
    getPlatformsEmptyQueryMock
];

const errorMocks = [
    getPlatformsErrorQueryMock
];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MockedProvider mocks={mocks} addTypename={false}><PlatformsComponent /></MockedProvider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shows empty message', async () => {
    const element = renderer
        .create(<MockedProvider mocks={emptyMocks} addTypename={false}><PlatformsComponent /></MockedProvider>);
    await wait(1);
    expect(element.toJSON()).toMatchSnapshot();
});

it('shows error message', async () => {
    const element = renderer
        .create(<MockedProvider mocks={errorMocks} addTypename={false}><PlatformsComponent /></MockedProvider>);

    await wait(1);
    expect(element.toJSON()).toMatchSnapshot();
});
