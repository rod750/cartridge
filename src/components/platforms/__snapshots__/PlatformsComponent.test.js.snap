// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows empty message 1`] = `
<div
  className="notification is-dark"
>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-alert"
    />
  </span>
  Uh oh. Something is not working.
</div>
`;

exports[`shows error message 1`] = `
<div
  className="notification is-dark"
>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-alert"
    />
  </span>
  Uh oh. Something is not working.
</div>
`;
